activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1138,6 +1158,31 @@ def preprocess_order_args(order_args)
         end.flatten!
       end
 
+      # If guard_unsafe_raw_sql is enabled, we only allow column names and
+      # directions as arguments to #order and #reorder. Other arguments will
+      # cause an ArugmentError to be raised.
+      def restrict_order_args(args)
+        return unless ActiveRecord::Base.guard_unsafe_raw_sql
+
+        columns = []
+        directions = []
+
+        hash = args.last.is_a?(Hash) ? args.pop : {}
+        directions.concat(hash.values)
+        columns.concat(hash.keys)
+        columns.concat(args)
+
+        bad_columns = columns.map(&:to_s) - klass.effective_column_names
+        if bad_columns.any?
+          raise ArgumentError, "Invalid order column: #{bad_columns}"
+        end
+
+        bad_directions = directions.map(&:to_s) - %w(asc desc ASC DESC)
```

Comment:
Rather than assign `directions` above, it's clear to ask for `orderings.values` here.